### PR TITLE
Refactoring preparations for relations

### PR DESCRIPTION
**Pull request by delgod**
in this PR:
* `README.md` and `CONTRIBUTING.md`
* config option and relations are renamed.
* renamed fields inside internal config structure.
* peer relation and config properties made public for future usage in the next PR

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## Prepare environment
+
+```shell
+# Juju
+sudo snap install juju --classic
+sudo usermod -a -G lxd $USER
+lxd init --auto
+lxc network set lxdbr0 ipv6.address none
+
```
Comment (jnsgruk) [inline CONTRIBUTING.md:30]: LXD isn't needed for the k8s charm :)

File: README.md [hunk 2]
```diff
@@ -1,34 +1,16 @@
...
-The charmed operator for [MongoDB](https://www.mongodb.com/) provides a general
-purpose distributed document database. This repository contains a
-[Juju](https://jaas.ai/) Charm for deploying MongoDB on Kubernetes
-clusters.
+Fully supported solution for production-grade MongoDB on Kubernetes from Canonical.
 
+Run the developer's favourite document database - MongoDB! Charm for MongoDB is a
+fully supported, automated solution from Canonical for running production-grade
+MongoDB on Kubernetes. It offers simple, secure and highly available setup with
+automatic recovery on failover. The solution includes scaling and other capabilities.
```
Comment (jnsgruk) [inline README.md:10]:
```suggestion
automatic recovery and failover. The solution includes scaling and other capabilities.
```

?

File: config.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # See LICENSE file for licensing details.
 
 options:
-  autodelete:
+  auto-delete:
     type: boolean
     description: Delete database if relations are removed
```
Comment (jnsgruk) [inline config.yaml:7]:
```suggestion
    description: |
        When a relation is removed, auto-delete ensures that any relevant databases 
        associated with the relation are also removed
```

**Review by jnsgruk — CHANGES_REQUESTED** [review]

File: src/charm.py [hunk 4]
```diff
@@ -33,7 +33,7 @@
 from tenacity import before_log, retry, stop_after_attempt, wait_fixed
 
 logger = logging.getLogger(__name__)
-PEER = "mongodb"
+PEER = "database-peers"
```
Comment (MiaAltieri) [inline src/charm.py:36]: Just curious what motivated this name change?

File: metadata.yaml [hunk 5]
```diff
@@ -10,11 +10,11 @@ description: |
...
 summary: A MongoDB operator charm for Kubernetes
 peers:
-  mongodb:
-    interface: mongodb
+  database-peers:
+    interface: mongodb-peers
 provides:
   database:
-    interface: mongodb_datastore
+    interface: mongodb-client
```
[the remark this answers is not in the document]
Comment (MiaAltieri) [reply metadata.yaml:17]: As of now the machine charm doesn't provide an interface

[the remark this answers is not in the document]
Comment (jnsgruk) [reply metadata.yaml:17]: The existing charm that bootstack maintained does :)

File: README.md (hunk 2 above)
Comment (delgod) [reply README.md:10]: wording can be better, automatic traffic switch needed on failover and automatic DB files recovery is needed during next start after HW fail.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+### Deploy
+
+```bash
+# Create a model
+juju add-model dev
+# Enable DEBUG logging
+juju model-config logging-config="<root>=INFO;unit=DEBUG"
+# Deploy the charm
+juju deploy ./mongodb_ubuntu-20.04-amd64.charm --resource mongodb-image=mongo:4.4 --num-units=1
```
[the remark this answers is not in the document]
Comment (delgod) [reply CONTRIBUTING.md:73]: I think the name is correct 
```
% charmcraft pack
Charms packed:
    mongodb_ubuntu-20.04-amd64.charm
```

File: README.md [hunk 7]
```diff
@@ -38,7 +20,7 @@ Create a Juju model for your operators, say "lma"
 
 Deploy a single unit of MongoDB using its default configuration
 
-    juju deploy ./mongodb_ubuntu-20.04-amd64.charm --resource mongodb-image=mongo:4.4
+    juju deploy ./mongodb_ubuntu-20.04-amd64.charm --resource mongodb-image=mongo:4.4 --num-units=1
```
[the remark this answers is not in the document]
Comment (delgod) [reply README.md:23]: I think the name is correct 
```
% charmcraft pack
Charms packed:
    mongodb_ubuntu-20.04-amd64.charm
```

File: config.yaml (hunk 3 above)
Comment (delgod) [reply config.yaml]: fixed in https://github.com/delgod/mongodb-k8s-operator/pull/2/commits/f1f3003da6928606039362b9be89055f5a2c77d2

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jnsgruk) [reply README.md:23]: Ah well in that case, the `metadata.yaml` is wrong ;-) The `metadata.yaml` file should state `name: mongodb-k8s` :)

File: README.md (hunk 2 above)
Comment (jnsgruk) [reply README.md:10]: Ok sure, let's leave this for now.

File: CONTRIBUTING.md (hunk 1 above)
Comment (delgod) [reply CONTRIBUTING.md]: right, moved to charmcraft section, it uses LXD for building charms https://github.com/delgod/mongodb-k8s-operator/pull/2/commits/b73a1a992fd2669bc80941ff5959db5356ec6c60

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (delgod) [reply README.md:23]: fixed in https://github.com/delgod/mongodb-k8s-operator/pull/2/commits/5a8f9b8ebdad7948d71b10739bc2215820fd101e

File: metadata.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (delgod) [reply metadata.yaml:17]: relation names of all charms were checked by @marceloneppel

File: CONTRIBUTING.md (hunk 1 above)
Comment (jnsgruk) [reply CONTRIBUTING.md]: Ah yes, indeed. My mistake.

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,4 @@
+redeploy:
```
Comment (jnsgruk) [inline Makefile:1]: Not sure I'd necessarily include this in the repo?

File: metadata.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (jnsgruk) [reply metadata.yaml:17]: See: https://git.launchpad.net/charm-mongodb/tree/metadata.yaml#n28

File: Makefile (hunk 8 above)
Comment (delgod) [reply Makefile]: fixed https://github.com/delgod/mongodb-k8s-operator/pull/2/commits/359d7938b523158e9ae9015730013006592b2a96

File: metadata.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (delgod) [reply metadata.yaml:17]: added as "obsolete" interface

**Review by jnsgruk — APPROVED** [review]

File: metadata.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (marceloneppel) [reply metadata.yaml:17]: I made a mistake assuming the relation name in the code of this github repository as mongodb_datastore (it's only in the edge channel on charmhub; the stable version there uses mongodb as you added here). I'll confirm every of our database charms on charmhub stable channel to confirm the right relation names with what we checked before in the github repositories.

**Review by MiaAltieri — APPROVED** [review]

**Pull request merged**